Move symbol layer properties to its section (#4565)
qgis/QGIS-Documentation: source/docs/user_manual/style_library/symbol_selector.rst

* the |dataDefined| :sup:`data-defined override` widget near almost all options,
  extending capabilities of customizing each symbol (see :ref:`data_defined` for
  more information)
* the |checkbox| :guilabel:`Enable layer` option controls the symbol layer's
  visibility. Disabled symbol layers are not drawn when rendering the symbol but
  are saved in the symbol. Being able to hide symbol layers is convenient when
  looking for the best design of your symbol as you don't need to remove any for

>   the testing. Also, with the data-defined property, this makes it easier to

Reviewer: ```suggestion
  the testing. Data-defined overrides then makes it possible to
```

Author: why the plural? We are referring to a particular property here. I'm also asking because of `make` vs `makes`

Reviewer: If we should go for singular, we would have to use either "A data-defined override" or "The data-defined override". Using plural makes the sentence a bit more "general".

Author: I don't want to be general, here. There's a data-defined override section for that. ;)
I  wanted to show how this feature can be extended beyond simple testing of symbol creation. The idea is you take a layer, you build a symbol with many symbol layers (eg, a circle, a square, a plane). Before this feature, if you did not want to display the circle for all the features of the layer, you'd need to create two symbols (`circle+square+plane` and `square+plane`) and assign each to the candidates. With this feature, you keep the trio and only DD the circle, so in your map, you display different symbol layers based on the features

>   display different symbol layers based on expression (eg, feature attributes).

Reviewer: ```suggestion
  display different symbol layers based on expressions (using, for instance, feature attributes).
```
